leader/leader.go

```diff
@@ -146,6 +154,7 @@ func (l *Client) AddWatch(key string, retry time.Duration, valuesC chan string)
 
 		watcher, resp, err = l.getWatchAtLatestIndex(ctx, api, key, retry)
 		if err != nil {
+			logger.WithError(err).Error("Failed to get watch at the latest index.")
 			return
```

Comment:
Is it correct behaviour to return here? I'm not sure if I'm missing something, but won't this basically cause the watch to die if the first attempt to setup the watch fails, whereas later iterations appear to retry? It might be meant to protect against a configuration error, but because AddWatch doesn't return an error I'm not sure how calling code is supposed to react to that.

Reply:
I concur with Kevin here - need to either return an error from `AddWatch` or leave it running.

The code doing the initial fetch can just move outside of the goroutine and be a prerequisite for it - and `AddWatch` needs to return an error.

Reply:
I had the same thought but if you look at how it's implemented, it actually never returns an error: getFirstValue that's called from getWatchAtLatestIndex always either returns a value, or sits in a loop until it gets it. Not sure why these methods return errors at all if they're always nil by current design - I'll update to remove them.